src/Domain/CoordinateMaps/CylindricalEndcap.cpp

```diff
@@ -0,0 +1,914 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "Domain/CoordinateMaps/CylindricalEndcap.hpp"
+
+#include <boost/none.hpp>
+#include <cmath>
+#include <gsl/gsl_poly.h>
+#include <limits>
+#include <pup.h>
+
+#include "DataStructures/Tensor/Tensor.hpp"
+#include "NumericalAlgorithms/RootFinding/QuadraticEquation.hpp"
+#include "Parallel/PupStlCpp11.hpp"
+#include "Utilities/ConstantExpressions.hpp"
+#include "Utilities/DereferenceWrapper.hpp"
+#include "Utilities/EqualWithinRoundoff.hpp"
+#include "Utilities/GenerateInstantiations.hpp"
+#include "Utilities/MakeWithValue.hpp"
+
+namespace domain::CoordinateMaps {
+
+namespace CylindricalEndcap_detail {
+
+// Maps a unit disk to the x>x_plane portion of a sphere
```

Comment:
If I'm reading this correctly, this is a 2D map from a disk to a portion of a spherical surface?

Reply:
Also, the unit disk lives on the x-y plane, but it seems the portion of the sphere lives roughly on the y-z plane. This would just be a style choice for me but I would keep x and y as the coordinates for both. In other words, map the disk to the `z>z_plane` portion of the sphere.

Reply:
Yeah, I too was thinking of making the output of this map oriented along the z-direction. Not sure if that should be done in this PR or the generalization...